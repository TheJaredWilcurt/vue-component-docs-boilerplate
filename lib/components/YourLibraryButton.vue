<template>
  <button
    :aria-pressed="active"
    :class="[
      'btn',
      'btn-' + status,
      { active }
    ]"
    data-test="button"
    @click="$emit('click', $event)"
  >
    <slot></slot>
  </button>
</template>

<script>
export default {
  name: 'YourLibraryButton',
  emits: ['click'],
  props: {
    active: {
      type: Boolean,
      default: false
    },
    status: {
      type: String,
      default: 'primary',
      validator: function (value) {
        return [
          'primary',
          'secondary',
          'info',
          'success',
          'warning',
          'danger'
        ].includes(value);
      }
    }
  }
};
</script>

<style>
.btn {
  border: 0px;
}
</style>
