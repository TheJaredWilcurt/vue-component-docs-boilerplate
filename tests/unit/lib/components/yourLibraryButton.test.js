import YourLibraryButton from '@/components/YourLibraryButton.vue';

import testHelpers from '@@/unit/testHelpers.js';

describe('YourLibraryButton.vue', () => {
  // Props
  const active = true;
  const status = 'success';
  const requiredProps = {};

  const setupWrapper = async (props) => {
    const options = {
      props: {
        ...requiredProps,
        ...props
      },
      slots: {
        default: 'Button Text'
      }
    };
    const wrapper = await testHelpers.mount(YourLibraryButton, options);
    return wrapper;
  };

  test('Renders correctly', async () => {
    const wrapper = await setupWrapper();

    expect(wrapper)
      .toMatchSnapshot();
  });

  describe('Props', () => {
    test('Button is active', async () => {
      const wrapper = await setupWrapper({ active });

      expect(wrapper)
        .toMatchSnapshot();
    });

    test('Button status is set', async () => {
      const wrapper = await setupWrapper({ status });

      expect(wrapper)
        .toMatchSnapshot();
    });
  });

  describe('Emits', () => {
    test('Emits click event', async () => {
      const wrapper = await setupWrapper();

      const button = wrapper.find('[data-test="button"]');
      await button.trigger('click');

      expect(wrapper.emitted().click[0].length)
        .toEqual(1);
    });
  });
});
