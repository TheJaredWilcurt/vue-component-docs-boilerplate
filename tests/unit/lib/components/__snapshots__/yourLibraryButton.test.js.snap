// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`YourLibraryButton.vue > Props > Button is active 1`] = `
<button
  aria-pressed="true"
  class="btn btn-primary active"
>
  Button Text
</button>
`;

exports[`YourLibraryButton.vue > Props > Button status is set 1`] = `
<button
  aria-pressed="false"
  class="btn btn-success"
>
  Button Text
</button>
`;

exports[`YourLibraryButton.vue > Renders correctly 1`] = `
<button
  aria-pressed="false"
  class="btn btn-primary"
>
  Button Text
</button>
`;
